src/apiCalls.js

```diff
@@ -0,0 +1,21 @@
+const handleError = (response) => {
+  if(response.ok) {
+    return response.json()
+  } else {
+    throw new Error(`HTTP Error: ${response.status} -- Please try again`)
+  }
+} 
+
+const getAllMovies = async () => {
+  const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+  const data = await handleError(response)
+  return data.movies
+}
+
+const getSingleMovie = async (id) => {
+  const response = await fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+  const data = await handleError(response)
+  return data.movie
+}
+
```

Comment:
To make this more dynamic, is there a way to consolidate the network requests into a single function instead of having two separate functions?